flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkBaseTest.java

```diff
@@ -412,6 +414,21 @@ public void testDoesNotWaitForPendingRequestsIfFlushingDisabled() throws Excepti
 		testHarness.close();
 	}
 
+	@Test
+	public void testOpenAndCloseInSinkFunction() throws Exception {
+		@SuppressWarnings("unchecked")
+		ElasticsearchSinkFunction<String> sinkFunction = (ElasticsearchSinkFunction<String>) mock(ElasticsearchSinkFunction.class);
+		final DummyElasticsearchSink<String> sink = new DummyElasticsearchSink<>(
+				new HashMap<>(), sinkFunction, new DummyRetryFailureHandler());
+
+		sink.open(mock(Configuration.class));
+		sink.close();
+
+		verify(sinkFunction, times(1)).open();
+		verify(sinkFunction, times(1)).close();
+		verifyNoMoreInteractions(sinkFunction);
```

Comment:
The community tends to avoid `mock` if possible. In this case, I think we could add `Testing ElasticsearchSinkFunction` or just an anonymous class.

Reply:
Sure I can update the test.

Out of curiosity, any particular reason to avoid `mock` whenever possible? In this case adding a simple implementation class would serve the exact same purpose as the mock, only lengthier.

Reply:
I think this reflects how well our design adheres to the principles of dependency injection. If our design is good enough, we could directly use the constructor in the test.